sdcm/sct_runner.py

```diff
@@ -589,9 +601,127 @@ def _get_base_image(self, image: Optional[Any] = None) -> Any:
+                        return gallery_image_version
+        return None
+
+    def _create_instance(self,  # pylint: disable=too-many-arguments
+                         instance_type: str,
+                         base_image: Any,
+                         tags: dict[str, str],
+                         instance_name: str,
+                         region_az: str = "",
+                         test_duration: Optional[int] = None) -> Any:
+        if base_image is self.BASE_IMAGE:
+            azure_region = self.azure_region_source
+            additional_kwargs = dict(
+                os_state=AzureOsState.GENERALIZED,
+                computer_name=self.image_name.replace(".", "-"),
+                admin_username=self.LOGIN_USER,
+                admin_public_key=self.key_pair.public_key.decode(),
+            )
+        else:
+            azure_region = self.azure_region
+            additional_kwargs = dict(
+                os_state=AzureOsState.SPECIALIZED,
+            )
+        return azure_region.create_virtual_machine(
+            vm_name=instance_name,
+            vm_size=instance_type,
+            image=base_image,
+            disk_size=self.instance_root_disk_size(test_duration=test_duration),
+            tags=tags | {"launch_time": get_current_datetime_formatted()},
+            **additional_kwargs,
+        )
+
+    def _stop_image_builder_instance(self, instance: Any) -> None:
+        self.azure_region_source.deallocate_virtual_machine(vm_name=instance.name)
+
+    def _terminate_image_builder_instance(self, instance: Any) -> None:
+        self.azure_service.delete_virtual_machine(virtual_machine=instance)
+
+    def _get_instance_id(self, instance: Any) -> Any:
+        return instance.id
+
+    def get_instance_public_ip(self, instance: Any) -> str:
+        return self.azure_service.get_virtual_machine_ips(virtual_machine=instance).public_ip
+
+    def _create_image(self, instance: Any) -> Any:
+        self.azure_region.create_gallery_image(
+            gallery_image_name=self.GALLERY_IMAGE_NAME,
+            os_state=AzureOsState.SPECIALIZED,
+        )
+        self.azure_region.create_gallery_image_version(
+            gallery_image_name=self.GALLERY_IMAGE_NAME,
+            gallery_image_version_name=self.GALLERY_IMAGE_VERSION,
+            source_id=instance.id,
+            tags=self.image_tags,
+        )
+
+    def _get_image_id(self, image: Any) -> Any:
+        return image.id
+
+    def _copy_source_image_to_region(self) -> None:
+        self.azure_region.append_target_region_to_image_version(
+            gallery_image_name=self.GALLERY_IMAGE_NAME,
+            gallery_image_version_name=self.GALLERY_IMAGE_VERSION,
+            region_name=self.region_name,
+        )
+
+    def _get_base_image(self, image: Optional[Any] = None) -> Any:
+        if image is None:
+            image = self.image
+        if isinstance(image, GalleryImageVersion):
+            return dict(id=image.id)
```

Comment:
It is not recommended to use dict class. "dict comprehension" is preferred one.

Reply:
Let me be a little bit nerdy here: you're actually asking for using dict literal, not dict comprehension (https://www.python.org/dev/peps/pep-0274/) :))

Pylint 2.10 added a new check `use-dict-literal` but for `dict()` vs `{}` case specifically, and the only reason to prefer dict literal over dict constructor I see is the performance (dict literal is 2x faster)

Can you point me to any document/article with better explanation? It'll be good to add this thing to our "Best practices."

But anyway I'll change it, thanks for raising this question :)

Reply:
Right, the point is in the performance. In my case the difference is bigger than 3 times.
It is the same as if we want to create some tmp loop with fixed set of elements we would better use "tuple" instead of "list" object types due to the same reasons - performance.
Sure, it is not game changed. But if we can improve something not losing anything, why not?